ConfigSpace/hyperparameters.pyx

```diff
@@ -326,18 +323,31 @@ cdef class FloatHyperparameter(NumericalHyperparameter):
         super(FloatHyperparameter, self).__init__(name, default_value, meta)
 
     def is_legal(self, value: Union[int, float]) -> bool:
-        raise NotImplemented
+        raise NotImplementedError()
 
     cpdef bint is_legal_vector(self, DTYPE_t value):
-        raise NotImplemented
+        raise NotImplementedError()
 
     def check_default(self, default_value: Union[int, float]) -> float:
-        raise NotImplemented
+        raise NotImplementedError()
 
+    def _transform(self, vector: Union[np.ndarray, float, int]) -> Optional[Union[np.ndarray, float, int]]:
+        if isinstance(vector, np.ndarray):
```

Comment:
Is there a reason to not use `_check_vector` here?

Reply:
The `_transform_vector()` and `_transform_scalar()` methods are now defined as cpdef's and check if the input is finite by themselves, i.e. accelerated by Cython. They raise a `ValueError` if the input is not finite. The `_transform()`  method here should instead catch the error and return None.